textproc/dblatex/Portfile

```diff
@@ -56,41 +56,66 @@ destroot.destdir    --root=${destroot} \
 # and installs dblatex's stylefiles to MacTeX's texmf (local)
 set dblatex.texmflocal ""
 set dblatex.mactex_bin ""
-
-variant mactex description {Allow dblatex to use a MacTeX installation instead of Macports texlive tools.
-To install the mactex variant /Library/TeX/texbin must be added to binpath
-in macports.conf} {
-    # First, check if MacTeX actually seems to be there…
-    set dblatex.mactex_candidates { \
-       "/Library/TeX/texbin" \
-       "/usr/texbin" \
+set dblatex.temp       ""
+
+variant mactex description {Allow dblatex to use MacTeX or another\
+    external TeXLive installation instead of MacPorts texlive tools.\
+    To install the 'mactex' variant the path to the TeX distribution's\
+    binary directory (for example '/Library/TeX/texbin') must be added\
+    to 'binpath' in 'macports.conf' *before* installing this port.} {
+
+    # Find the binary directory of the external TeX distribution by
+    # searching the kpsewhich program in the path.  We assume that all
+    # other binaries of the distribution are in this directory, too.
+    if {[auto_execok kpsewhich] ne ""} {
+        set dblatex.temp {*}[auto_execok kpsewhich]
     }
-    foreach dir ${dblatex.mactex_candidates} {
-        if [file executable "${dir}/kpsewhich"] {
-            set dblatex.mactex_bin ${dir}
-            break
+    set dblatex.mactex_bin \
+        [file dirname [file normalize ${dblatex.temp}]]
+    if {${dblatex.mactex_bin} eq "."} {
+        pre-fetch {
+            return -code error "Cannot find MacTeX or external TeXLive\
+                installation; aborting.  Have you added the path to\
+                your TeX distribution's binary directory to 'binpath'\
+                in 'macports.conf'?"
         }
-    }
-    if { ${dblatex.mactex_bin} != "" } {
+    } elseif {${dblatex.mactex_bin} eq \
+                  [file dirname [file normalize "${prefix}/bin/kpsewhich"]]} {
+        pre-fetch {
+            return -code error "Variant 'mactex' doesn't work with\
+                installed macports texlive packages.  Either\
+                uninstall them or don't use the 'mactex' variant of\
+                dblatex."
+        }
+    } else {
         set dblatex.texmflocal \
             [exec ${dblatex.mactex_bin}/kpsewhich --expand-var='\$TEXMFLOCAL']
         regsub -all {'} ${dblatex.texmflocal} "" dblatex.texmflocal
+    }
+
+    # Make 'port notes dblatex +mactex' work even if neither the package
+    # nor an external TeXLive is installed.
+    if {${dblatex.texmflocal} eq ""} {
+        notes "The 'mactex' variant will install styles into your\
+            TeXLive's texmf-local tree (which is outside MacPort's\
```

Comment:
Our name is "MacPorts" not "MacPort" so change `MacPort's` to either `MacPorts'` or `the MacPorts`.